modules/build/src/main/scala/scala/build/internal/AmmUtil.scala

```diff
@@ -18,5 +20,9 @@ object AmmUtil {
 
   def encodeScalaSourcePath(path: Seq[Name]) = path.map(_.backticked).mkString(".")
 
-  def normalizeNewlines(s: String) = s.replace("\r", "").replace("\n", System.lineSeparator())
+  def normalizeNewlines(s: String)(using newLine: String = lineSeparator): String =
+    s.replace("\r", "").replace("\n", newLine)
+
+  lazy val lineSeparator: String =
+    if EnvVar.Misc.shell.valueOpt.nonEmpty then "\n" else System.lineSeparator()
```

Comment:
Wait... `\n` for all shells, arbitrarily?
why?
shouldn't you handle specific cases here?

Reply:
I'd imagine smth like
```suggestion
EnvVar.Misc.shell.valueOpt.map(_.split("[\\/]+").last).map {
          case ??? => "\n" // for whichever shell you'd like to handle
          case _  => System.lineSeparator()
        }.getOrElse(System.lineSeparator())
```

Reply:
Carriage return leads to errors:

- `^M: command not found`
- `/bin/bash^M: bad interpreter: No such file or directory`

 where the `\r` is interpreted as part of the command.

If SHELL is defined, 99% likelihood is `bash`, `sh`, `ksh`, `zsh` or `fish`, none of which expect or tolerate carriage returns.

We could instead make it conditional on [`isShebangCapableShell`](https://github.com/VirtusLab/scala-cli/blob/8298cc06686abcf8bbe190b11cc484271bed99bc/modules/cli/src/main/scala/scala/cli/internal/ProcUtil.scala#L102) but that would be less effective in avoiding these types of errors, since new shells might become popular in the future.

The only case I'm aware of in which a Carriage Return is required is when passing command strings to `CMD.EXE`.  Otherwise, there it's not required, even in Windows (my experience).

Reply:
...shouldn't we at least handle `cmd` here, then?

Reply:
The `AmmUtil.lineSeparator` change only affects line endings of generated wrapper code.  The line endings of wrapped client code aren't modified, so the semantics of client code w.r.t. line endings sent to the console or written to files, etc. aren't affected.   

This is mostly a cosmetic issue, visible in some text editors (see #3756).

I did some experiments with a native-compiled `scala-cli`.   Running a script in a CMD.EXE terminal, with or without SHELL defined had no effect on how the code ran, or on the output, as expected.

However, in Windows, the following tests will fail if `System.lineSeparator` is used, since the tests compare exact literal Strings against expected values:
```scala
[1026] =[1026] ==> X scala.build.tests.markdown.MarkdownCodeWrapperTests.a simple Scala code block is wrapped correctly  0.032s java.lang.AssertionError: assertion failed
[1026] ==> X scala.build.tests.markdown.MarkdownCodeWrapperTests.multiple plain Scala code blocks are wrapped correctly  0.003s java.lang.AssertionError: assertion failed
[1026] ==> X scala.build.tests.markdown.MarkdownCodeWrapperTests.multiple plain Scala code blocks with different scopes are wrapped correctly  0.003s java.lang.AssertionError: assertion failed
[1026] ==> X scala.build.tests.markdown.MarkdownCodeWrapperTests.a raw Scala code block is wrapped correctly  0.001s java.lang.AssertionError: assertion failed
[1026] ==> X scala.build.tests.markdown.MarkdownCodeWrapperTests.multiple raw Scala code blocks are glued together correctly  0.01s java.lang.AssertionError: assertion failed
[1026] ==> X scala.build.tests.markdown.MarkdownCodeWrapperTests.a test Scala snippet is wrapped correctly  0.009s java.lang.AssertionError: assertion failed
[1026] ==> X scala.build.tests.markdown.MarkdownCodeWrapperTests.multiple test Scala snippets are glued together correctly  0.013s java.lang.AssertionError: assertion failed
```

Line endings depend on where a String originates:

- client source code: "\n" in a SHELL environment, unpredictable otherwise
- Triple quote strings:  taken from the enclosing source file

The `scala-cli` source files are normalized to "\n" on commit (by .gitattributes "* text=auto"), so tripled quoted strings in our Test sources will not have `\r` except in some `git` client configurations.

Reply:
I now understand even less on why do you rely on `EnvVar.Misc.shell` being defined here, however.
Shouldn't it then always be `\n`?

Reply:
> why do you rely on `EnvVar.Misc.shell`

The intent was that if testing from a `CMD.EXE` this causes it to revert to the previous behavior.   Not sure if anyone actually does that, because  some of the tests always fail in a Windows build, AFAIK.  It might be that most `scala-cli` Windows developers work in `WSL`.   

> Shouldn't it then always be `\n`

Your suggestion would certainly simplify things,  by eliminating `\r` from the equation in all cases.    I'll do some experiments.

Reply:
In a clone of this PR code (checked out with CRLF line endings), applied the suggested simplification, and tested in a git-for-windows bash session.   Because `EnvVar.Misc.shell` is no longer referenced, no need to attempt tests in a `CMD` session.   Tests pass in both git-for-windows and in `msys64` bash session with LF-only line endings.

Encountered and fixed #3725.